From bcfdaafde561a22874733d22962dad387d9f4e48 Mon Sep 17 00:00:00 2001
From: Ruediger Pluem <rpluem@apache.org>
Date: Fri, 10 Mar 2023 10:54:03 +0000
Subject: [PATCH] Revert: Fix crosscompiling detection in apr-1-config

Since r1872148 apr-1-config tries to detect if it runs in a crosscompile
scenario. Fix the detection to correctly distinguish between crosscompiling
and DESTDIR installations for packaging.

* configure.in: Set APR_CROSS_COMPILING to 'no', 'yes' or 'maybe' based on
      possible --host and --build parameters given to configure to determine
      if configure was called for a crosscompiling scenario. This was inspired
      by how autoconf itself detects crosscompiling scenarios.
* apr-config.in: Use the value from APR_CROSS_COMPILING to determine if
      we are in a crosscompiling scenario. Furthermore normalize the pathes
      we use for suffix matches by collapsing multiple consecutive '/' to one.

PR: 66510

git-svn-id: https://svn.apache.org/repos/asf/apr/apr/trunk@1908248 13f79535-47bb-0310-9956-ffa450edef68

[Revert upstream commit https://github.com/apache/apr/commit/130cf8cb3c6fe713095e560ade1df050aa2a6a14]
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 apr-config.in | 29 +++++++----------------------
 configure.in  | 12 ------------
 2 files changed, 7 insertions(+), 34 deletions(-)

diff --git a/apr-config.in b/apr-config.in
index 626d3b0..bed47ca 100644
--- a/apr-config.in
+++ b/apr-config.in
@@ -48,28 +48,13 @@ APR_LIBNAME="@APR_LIBNAME@"
 # NOTE: the following line is modified during 'make install': alter with care!
 location=@APR_CONFIG_LOCATION@
 
-cross_compiling=@APR_CROSS_COMPILING@
-
-if test "$cross_compiling" != "no"; then
-
-    # Normalize $0 and bindir by removing consecutive '/' as the comparison
-    # and the suffix removal below might fail for semantic equal pathes.
-    # XXX: This is not perfect. On Linux we could use realpath for this purpose
-    # but realpath is not portable.
-    normalized_command=`echo $0 | tr -s /`
-    normalized_bindir=`echo ${bindir} | tr -s /`
-    # Remove trailing '/'
-    normalized_bindir=${normalized_bindir%/}
-
-    # absolute path, but not installed path - we're cross compiling
-    case "$normalized_command" in
-      "${normalized_bindir}/"*) ;;
-      "/"*)                     location=crosscompile;
-                                APR_TARGET_DIR=${normalized_command%${normalized_bindir}/apr-${APR_MAJOR_VERSION}-config};
-                                ;;
-      *)                        ;;
-    esac
-fi
+# absolute path, but not installed path - we're cross compiling
+case "$0" in
+  "${bindir}/"*) ;;
+  "/"*)         location=crosscompile;
+                APR_TARGET_DIR=${0%${bindir}/apr-${APR_MAJOR_VERSION}-config} ;;
+  *)            ;;
+esac
 
 show_usage()
 {
diff --git a/configure.in b/configure.in
index ff92e55..d9898c5 100644
--- a/configure.in
+++ b/configure.in
@@ -140,18 +140,6 @@ fi
 
 AC_SUBST(APR_CONFIG_LOCATION)
 
-if test "x$host_alias" != x; then
-  if test "x$build_alias" = x; then
-    APR_CROSS_COMPILING=maybe
-  elif test "x$build_alias" != "x$host_alias"; then
-    APR_CROSS_COMPILING=yes
-  fi
-else
-    APR_CROSS_COMPILING=no
-fi
-
-AC_SUBST(APR_CROSS_COMPILING)
-
 # Libtool might need this symbol -- it must point to the location of
 # the generated libtool script (not necessarily the "top" build dir).
 #
-- 
2.45.2

